pkg/virt-handler/isolation/process_test.go

```diff
@@ -0,0 +1,119 @@
+	const (
+		processTestExecPath = "processA"
+		processTestPID      = 110
+		nonExistPPid        = 300
+	)
+	emptyProcessList := []ProcessStub{}
+	procStub1 := ProcessStub{ppid: 1, pid: 120, binary: processTestExecPath}
+	procStub2 := ProcessStub{ppid: processTestPID, pid: 2222, binary: "processB"}
+	procStub3 := ProcessStub{ppid: 1, pid: 110, binary: "processC"}
+	procStub4 := ProcessStub{ppid: processTestPID, pid: 3333, binary: "processD"}
+	testProcesses := []ProcessStub{procStub1, procStub3, procStub2, procStub4}
+
+	Context("find child processes", func() {
+		table.DescribeTable("should return the correct child processes of the given pid",
+			func(processes []ProcessStub, ppid int, expectedProcesses []ProcessStub) {
+				Expect(childProcesses(convertToPsGoProcessList(processes), ppid)).
+					To(ConsistOf(convertToPsGoProcessList(expectedProcesses)))
+			},
+			table.Entry("given no input processes, there are no child processes",
+				emptyProcessList, nonExistPPid, emptyProcessList,
+			),
+			table.Entry("given process list and non-exist pid, should return no child processes",
+				testProcesses, nonExistPPid, emptyProcessList,
+			),
+			table.Entry("given process list and pid where there are child processes of the given pid",
+				testProcesses, processTestPID, []ProcessStub{procStub2, procStub4},
+			),
+		)
+	})
+
+	Context("lookup process by executable", func() {
+		procStub5 := ProcessStub{ppid: 100, pid: 220, binary: processTestExecPath}
+
+		table.DescribeTable("should find no process",
+			func(processes []ProcessStub, executable string) {
+				Expect(lookupProcessByExecutable(convertToPsGoProcessList(processes), executable)).To(BeNil())
+			},
+			table.Entry("given no input processes and empty string as executable",
+				emptyProcessList, "",
+			),
+			table.Entry("given no input processes and executable",
+				emptyProcessList, "processA",
+			),
+			table.Entry("given processes list and empty string",
+				testProcesses, "",
+			),
+		)
+
+		table.DescribeTable("should return the first occurrence of a process that runs the given executable",
+			func(processes []ProcessStub, executable string, expectedProcess ps.Process) {
+				Expect(lookupProcessByExecutable(convertToPsGoProcessList(processes), executable)).
+					To(Equal(expectedProcess))
+			},
+			table.Entry("given processes list that includes exactly one process that runs the executable",
+				testProcesses, processTestExecPath, &procStub1,
+			),
+			table.Entry("given processes list that includes more than one process that runs the executable",
+				append(testProcesses, procStub5), processTestExecPath, &procStub1,
+			),
+		)
+	})
+})
+
+func convertToPsGoProcessList(processes []ProcessStub) []ps.Process {
+	result := []ps.Process{}
+	for idx := range processes {
+		result = append(result, &processes[idx])
+	}
+
+	return result
+}
+
+type ProcessStub struct {
+	ppid   int
+	pid    int
+	binary string
+}
+
+func (p *ProcessStub) Pid() int {
```

Comment:
I followed how `ps-go` package implements other kinds of processes (e.g `UnixProcess`) where the receivers were pointers.
I can confirm that if the receivers are pointers then its not possible to define `ps.Process` array where the objects are from a different type (e.g `ProcessStub`).
Removing the pointers from the receivers makes it possible, I dont know to explain why though.

Reply:
If the receiver is a pointer, you can populate the list with both pointers and values: https://play.golang.org/p/mVnwoCUmGT_q

If the receiver is a value, you can no longer use values in the list, you need pointers.

Reply:
@maiqueb  I tested it and its like you said, Go compiler didn't recognized `ProcessStub` as implementation of `ps.Process`
because the receiver type of `ProcessStub` is pointer type. 
But on the tests `ProcessStub` objects are defined as non-pointer types.
I also removed the pointers from `ProcessStub` as we discussed there is no actual need for them to be pointers.
Thanks again for the heads up!